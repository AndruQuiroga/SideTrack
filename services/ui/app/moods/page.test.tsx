import { render } from '@testing-library/react';
import Moods from './page';

describe('Moods page', () => {
  beforeEach(() => {
    global.fetch = jest.fn((url: RequestInfo) => {
      if (typeof url === 'string' && url.includes('/dashboard/trajectory')) {
        return Promise.resolve({
          ok: true,
          json: async () => ({ points: [{ week: '2024-06-17' }, { week: '2024-06-24' }] }),
        }) as any;
      }
      if (typeof url === 'string' && url.includes('/dashboard/radar')) {
        const u = new URL(url);
        const week = u.searchParams.get('week');
        return Promise.resolve({
          ok: true,
          json: async () => ({
            week,
            axes: { energy: 0.1, valence: 0.2, danceability: 0.3, brightness: 0.4, pumpiness: 0.5 },
          }),
        }) as any;
      }
      return Promise.resolve({ ok: false }) as any;
    }) as any;
  });

  afterEach(() => {
    (global.fetch as jest.Mock).mockReset();
  });

  it('renders mood shares chart', async () => {
    const { container } = render(await Moods());
    expect(container).toMatchSnapshot();
  });
});
