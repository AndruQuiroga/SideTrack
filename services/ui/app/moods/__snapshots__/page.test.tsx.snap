// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods page renders mood shares chart 1`] = `
<div>
  <section>
    <h2>
      Moods
    </h2>
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 600px; height: 300px;"
    >
      <svg
        class="recharts-surface"
        height="300"
        style="width: 100%; height: 100%;"
        viewBox="0 0 600 300"
        width="600"
      >
        <title />
        <desc />
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="260"
              width="530"
              x="65"
              y="5"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="595"
              y="5"
              y1="265"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="595"
              y="5"
              y1="200"
              y2="200"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="595"
              y="5"
              y1="135"
              y2="135"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="595"
              y="5"
              y1="70.00000000000003"
              y2="70.00000000000003"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="595"
              y="5"
              y1="5"
              y2="5"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="65"
              x2="65"
              y="5"
              y1="5"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="530"
              x="65"
              x1="595"
              x2="595"
              y="5"
              y1="5"
              y2="265"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            orientation="bottom"
            stroke="#666"
            width="530"
            x="65"
            x1="65"
            x2="595"
            y="265"
            y1="265"
            y2="265"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                width="530"
                x="65"
                x1="65"
                x2="65"
                y="265"
                y1="271"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                width="530"
                x="65"
                y="273"
              >
                <tspan
                  dy="0.71em"
                  x="65"
                >
                  2024-06-17
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                width="530"
                x="65"
                x1="595"
                x2="595"
                y="265"
                y1="271"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                width="530"
                x="595"
                y="273"
              >
                <tspan
                  dy="0.71em"
                  x="595"
                >
                  2024-06-24
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="260"
            orientation="left"
            stroke="#666"
            width="60"
            x="5"
            x1="65"
            x2="65"
            y="5"
            y1="5"
            y2="265"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="265"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                width="60"
                x="57"
                y="265"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="200"
                y2="200"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                width="60"
                x="57"
                y="200"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  0.4
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="135"
                y2="135"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                width="60"
                x="57"
                y="135"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  0.8
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="70.00000000000003"
                y2="70.00000000000003"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                width="60"
                x="57"
                y="70.00000000000003"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  1.2
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="5"
                y2="5"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                width="60"
                x="57"
                y="5"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  1.6
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-2"
              >
                <rect
                  height="266"
                  width="0"
                  x="65"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-2)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M65,248.75L595,248.75L595,265L65,265Z"
                  fill="#8884d8"
                  fill-opacity="0.6"
                  height="260"
                  stroke="none"
                  width="530"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M65,248.75L595,248.75"
                  fill="none"
                  fill-opacity="0.6"
                  height="260"
                  stroke="#8884d8"
                  width="530"
                />
              </g>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-3"
              >
                <rect
                  height="249"
                  width="0"
                  x="65"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-3)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M65,216.25L595,216.25L595,248.75L65,248.75Z"
                  fill="#82ca9d"
                  fill-opacity="0.6"
                  height="260"
                  stroke="none"
                  width="530"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M65,216.25L595,216.25"
                  fill="none"
                  fill-opacity="0.6"
                  height="260"
                  stroke="#82ca9d"
                  width="530"
                />
              </g>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-4"
              >
                <rect
                  height="217"
                  width="0"
                  x="65"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-4)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M65,167.5L595,167.5L595,216.25L65,216.25Z"
                  fill="#ffc658"
                  fill-opacity="0.6"
                  height="260"
                  stroke="none"
                  width="530"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M65,167.5L595,167.5"
                  fill="none"
                  fill-opacity="0.6"
                  height="260"
                  stroke="#ffc658"
                  width="530"
                />
              </g>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-5"
              >
                <rect
                  height="168"
                  width="0"
                  x="65"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-5)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M65,102.5L595,102.5L595,167.5L65,167.5Z"
                  fill="#ff7300"
                  fill-opacity="0.6"
                  height="260"
                  stroke="none"
                  width="530"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M65,102.5L595,102.5"
                  fill="none"
                  fill-opacity="0.6"
                  height="260"
                  stroke="#ff7300"
                  width="530"
                />
              </g>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-area"
        >
          <g
            class="recharts-layer"
          >
            <defs>
              <clippath
                id="animationClipPath-recharts-area-6"
              >
                <rect
                  height="103"
                  width="0"
                  x="65"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer"
              clip-path="url(#animationClipPath-recharts-area-6)"
            >
              <g
                class="recharts-layer"
              >
                <path
                  class="recharts-curve recharts-area-area"
                  d="M65,21.25L595,21.25L595,102.5L65,102.5Z"
                  fill="#a4de6c"
                  fill-opacity="0.6"
                  height="260"
                  stroke="none"
                  width="530"
                />
                <path
                  class="recharts-curve recharts-area-curve"
                  d="M65,21.25L595,21.25"
                  fill="none"
                  fill-opacity="0.6"
                  height="260"
                  stroke="#a4de6c"
                  width="530"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
      <div
        class="recharts-tooltip-wrapper"
        style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"
        tabindex="-1"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
